### PR TITLE
[GEOS-7644]: Make secure decorators supporting StructuredGridCoverage…

### PR DESCRIPTION
**Pull request by dromagnoli**
…2DReaders

### CODE REVIEW

File: src/main/src/main/java/org/geoserver/security/decorators/DefaultSecureDataFactory.java [hunk 1]
```diff
@@ -114,14 +115,16 @@ public Object secure(Object object, WrapperPolicy policy) {
         } else if (FeatureIterator.class.isAssignableFrom(clazz)) {
             return new SecuredFeatureIterator((FeatureIterator) object); 
         }
-        
+
         // try coverage readers and formats
         if(GridCoverage2DReader.class.isAssignableFrom(clazz)) {
             return new SecuredGridCoverage2DReader((GridCoverage2DReader) object, policy);
+        } else if(StructuredGridCoverage2DReader.class.isAssignableFrom(clazz)) {
```
Comment (aaime) [inline src/main/src/main/java/org/geoserver/security/decorators/DefaultSecureDataFactory.java]: Given that:

{code}
public interface StructuredGridCoverage2DReader extends GridCoverage2DReader {
{code}

Should't you test the structured one before testing its super-interface?

File: src/main/src/main/java/org/geoserver/security/decorators/SecuredStructuredGridCoverage2DReader.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+ */
+public class SecuredStructuredGridCoverage2DReader extends DecoratingStructuredGridCoverage2DReader {
+
+    /** Parameters used to control the {@link Crop} operation. */
+    private static final ParameterValueGroup cropParams;
+
+    /**
+     * Cached crop factory
+     */
+    private final static Crop coverageCropFactory = new Crop();
```
Comment (aaime) [inline src/main/src/main/java/org/geoserver/security/decorators/SecuredStructuredGridCoverage2DReader.java]: Is this crop factory still needed?

File: src/main/src/main/java/org/geoserver/security/decorators/DefaultSecureDataFactory.java (hunk 1 above)
Comment (dromagnoli) [inline src/main/src/main/java/org/geoserver/security/decorators/DefaultSecureDataFactory.java]: Right, this remembers me why I put the structured in first place 2 weeks ago when doing the local patch. Fixing it.

**Comment by aaime** [comment]
Looks good now, merging

**Pull request merged**